crates/ratchet-core/src/cpu/mod.rs

```diff
@@ -287,6 +287,64 @@ pub fn cpu_cast(cast: Cast, dst: Tensor) -> Result<Tensor, OperationError> {
     Ok(dst)
 }
 
+fn concat_inner<T: TensorDType>(
+    inputs: RVec<Tensor>,
+    dim: usize,
+    dst: Tensor,
+) -> Result<Tensor, OperationError> {
+    inputs.iter().try_for_each(|i| {
```

Comment:
This is checked on `OpGuards` no?

Reply:
`assert!(self.inputs.iter().all(|x| self.dim < x.rank()));`

It sure is!